### DISCUSSION TITLE
JSBSim for DCS (EFM based on JSBSim)

### DISCUSSION THREAD

**Discussion by thewhitetile** [JSBSim-Team/jsbsim#1024] [Ideas]
Before undertaking the task of using JSBSim to create an aircraft in DCS I wanted to see if anyone else has worked on this.  It looks like it would not be too hard.  It would take in input from a Hotas and then use JSBSim to get the forces and moments on the aircraft.  Then pass those back to DCS.  Is there anything similar to this that has been done that I can work from?

**Comment by seanmcleod** [comment]
I haven't used JSBSim with DCS but I have used it to provide a FDM (Flight Dynamics Model) for Prepar3D.

I assume you're looking at making use of the External Flight Model (EFM) option mentioned here - https://www.digitalcombatsimulator.com/en/support/faq/505/

Do you have a link to the API documentation for DCS's EFM?

**Comment by thewhitetile** [comment]
Sorry for the delay.  Here is a couple examples of EFMs.  The A-4 is the
best one.  The F-16 one is a much simpler example.

https://github.com/RagnarDa/F-16Demo/blob/master/F16DemoReadme.txt
https://github.com/heclak/community-a4e-c

Let me know what you find out.

**Comment by Zaretto** [comment]
I made an EFM that permits the use of a JSBSim model in DCS world directly. see https://github.com/Zaretto/acEFM 

There is a full example (without 3d models) here https://github.com/Zaretto/DCS-SEPECAT-Jaguar/tree/main
